execution_engine/src/core/engine_state/mod.rs

```diff
@@ -1173,7 +1173,8 @@ where
             Err(error) => return Ok(ExecutionResult::precondition_failure(error.into())),
         };
 
-        let max_payment_cost = Motes::new(*MAX_PAYMENT);
+        let max_payment_cost =
+            Motes::from_gas(*MAX_PAYMENT, 1u64).ok_or(Error::GasConversionOverflow)?;
```

Comment:
We need to use supplied gas price for this conversion

```suggestion
        let max_payment_cost =
            Motes::from_gas(*MAX_PAYMENT, deploy_item.gas_price).ok_or(Error::GasConversionOverflow)?;
```

Reply:
Applied suggestion